tools/src/main/scala/caliban/tools/RemoteSchema.scala

```diff
@@ -0,0 +1,313 @@
+      description = definition.description,
+      interfaces = toInterfaces(definition.implements, definitions),
+      directives = toDirectives(definition.directives),
+      fields = (args: __DeprecatedArgs) => {
+        if (definition.fields.length > 0)
+          Some(
+            definition.fields
+              .map(toField(_, definitions))
+              .filter(filterDeprecated(_, args))
+          )
+        else None
+      }
+    )
+
+  private def toField(
+    definition: Definition.TypeSystemDefinition.TypeDefinition.FieldDefinition,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ): __Field =
+    __Field(
+      name = definition.name,
+      description = definition.description,
+      args = definition.args.map(toInputValue(_, definitions)),
+      `type` = toType(definition.ofType, definitions),
+      isDeprecated = isDeprecated(definition.directives),
+      deprecationReason = deprecationReason(definition.directives),
+      directives = toDirectives(definition.directives)
+    )
+
+  private def toType(
+    definition: Type,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ): () => __Type = { () =>
+    definition match {
+      case Type.ListType(t, nonNull) =>
+        if (nonNull)
+          __Type(
+            kind = __TypeKind.NON_NULL,
+            ofType = Some(
+              __Type(
+                kind = __TypeKind.LIST,
+                ofType = Some(
+                  toType(t, definitions)()
+                )
+              )
+            )
+          )
+        else
+          __Type(
+            kind = __TypeKind.LIST,
+            ofType = Some(
+              toType(t, definitions)()
+            )
+          )
+
+      case Type.NamedType(name, nonNull) =>
+        if (nonNull)
+          __Type(
+            kind = __TypeKind.NON_NULL,
+            ofType = Some(
+              toType(name, definitions)
+            )
+          )
+        else
+          toType(name, definitions)
+    }
+  }
+
+  private def toType(
+    name: String,
+    definitions: List[Definition.TypeSystemDefinition.TypeDefinition]
+  ) =
+    definitions.find(_.name == name) match {
+      case Some(value) =>
+        value match {
+          case e: EnumTypeDefinition        => toEnumType(e)
+          case s: ScalarTypeDefinition      => toScalar(s)
+          case u: UnionTypeDefinition       => toUnionType(u, definitions)
+          case o: ObjectTypeDefinition      => toObjectType(o, definitions)
+          case i: InputObjectTypeDefinition =>
+            toInputObjectType(i, definitions)
+          case i: InterfaceTypeDefinition   =>
+            toInterfaceType(i, definitions)
+        }
+      case None        => __Type(kind = __TypeKind.SCALAR, name = Some(name))
+    }
+
+  private def toDirectives(directives: List[Directive]): Option[List[Directive]] =
+    if (directives.length > 0) Some(directives.filter(_.name != "deprecated"))
```

Comment:
Would we want to apply the `directive.nonEmpty` check _after_ the `filter`?

Reply:
Yes, that makes a lot more sense!